addons/web/static/src/core/debug/profiling/profiling_qweb.js

```diff
@@ -0,0 +1,331 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+import { loadBundle } from "@web/core/assets";
+import { renderToString } from "@web/core/utils/render";
+import { useDebounced } from "@web/core/utils/timing";
+
+const { Component, useState, useRef, onWillStart, onMounted, onWillUnmount } = owl;
+
+class MenuItem extends Component {}
+MenuItem.template = "web.ProfilingQwebView.menuitem";
+
+/**
+ * This widget is intended to be used on Text fields. It will provide Ace Editor
+ * for display XML and Python profiling.
+ */
+export class ProfilingQwebView extends Component {
+    setup() {
+        super.setup();
+
+        this.orm = useService("orm");
+        this.ace = useRef("ace");
+        this.selector = useRef("selector");
+        this.mode = "readonly";
+
+        this.profileLines = this.results.data;
```

Comment:
this makes the results getter useless, because even if the props change, this.profileLines won't be updated. 
You probably should just use `this.results.data` where you need it. If you want a better name, you can juste use the getter to return a renamed attribute.